### ISSUE TITLE
RRULE does not seem to process BY DAY correctly

### ISSUE THREAD

**Issue by lightsofapollo** [kewisch/ical.js#30]
In the process of writing tests for icalrecur I found some issues with by day rules

FREQ=WEEKLY;COUNT=10;WKST=SU;BYDAY=TU,TH
FREQ=YEARLY;UNTIL=20150131T090000Z;BYMONTH=1;BYDAY=SU,MO,TU,WE,TH,FR,SA

Working on it now, here for tracking

**Comment by lightsofapollo** [comment]
The Second rule for example with yield the following dates when given this start point:
20120101T090000

``` js
[ Sat, 14 Jan 2012 17:00:00 GMT,
  Sat, 21 Jan 2012 17:00:00 GMT,
  Sat, 28 Jan 2012 17:00:00 GMT,
  Sat, 07 Jan 2012 17:00:00 GMT,
  Sat, 12 Jan 2013 17:00:00 GMT,
  Sat, 19 Jan 2013 17:00:00 GMT,
  Sat, 26 Jan 2013 17:00:00 GMT,
  Sat, 05 Jan 2013 17:00:00 GMT,
  Sat, 11 Jan 2014 17:00:00 GMT,
  Sat, 18 Jan 2014 17:00:00 GMT,
  Sat, 25 Jan 2014 17:00:00 GMT,
  Sat, 04 Jan 2014 17:00:00 GMT,
  Sat, 10 Jan 2015 17:00:00 GMT,
  Sat, 17 Jan 2015 17:00:00 GMT,
  Sat, 24 Jan 2015 17:00:00 GMT,
  Sat, 31 Jan 2015 17:00:00 GMT,
  Sat, 03 Jan 2015 17:00:00 GMT ]
```

**Comment by lightsofapollo** [comment]
Taken from http://tools.ietf.org/html/rfc5545#page-124

```
Every day in January, for 3 years:
.....

       RRULE:FREQ=YEARLY;UNTIL=20000131T140000Z;BYMONTH=1;BYDAY=SU,MO,TU,WE,TH,FR,SA
```

**Comment by lightsofapollo** [comment]
Ok I have tracked the problem down.

Per http://tools.ietf.org/html/rfc5545#page-42 BYDAY (without a numeric part) in FREQ = YEARLY / MONTHLY means each weekday in that month/year. It seems like at minimum we are not doing this. The code path in ICAL.icaltime#nth_week_day also had some issues when asking for a weekday that falls outside of its given month.

In addition I have found nth_week_day always resets to the start of the month in the negative case rather then starting from its given time ( which I believe? is the correct route)
